src/math.hpp

```diff
@@ -5,79 +5,65 @@
 
 #include <cmath>
 #include <cstddef>
+#include <iostream>
 #include <type_traits>
 
 namespace opt {
 
-template <Arithmetic T>  // clang-format off
-  requires(not Dual<T>)
-constexpr auto exp(T x)  // clang-format on
+template <class T>  // clang-format off
+  requires(Arithmetic<T> || Dual<T>)
```

Comment:
template `<Arithmetic T>`
is sufficient a `Dual` is also an `Arithmetic`

Reply:
Done